### ISSUE TITLE
cannot set latency (latency combo box empty)

### ISSUE THREAD

**Issue by Holzhaus** [Holzhaus/mixxx-gh-issue-migration#359]
I can't set the latency, because the combobox is empty.
OS: Windows XP SP3
CPU: AMD Turion 64
Soundcard: Audio2DJ
Videocard: ATI Radeon Xpress 1150

At the first run the combobox was not empty. I tried to change the value and clicked apply and/or ok. Next time, when i opened the Preferences dialogbox, the list was empty. (And it is still empty.)

**Comment by Holzhaus** [comment]
(by sandor-ivicsics)

**Comment by Holzhaus** [comment]
(by sandor-ivicsics)
Mixxx 1.9.0_beta

**Comment by Holzhaus** [comment]
(by rryan)
I've seen this too. A workaround is to change the samplerate. This triggers an update of the Latency combo-box.

**Comment by Holzhaus** [comment]
(by bkgood)
I'm supposed to be on mixxx hiatus for exams but a possible cause/solution to this bug occurred to me when I saw this bugmail. A potential fix is in lp:mixxx/1.9, no promises as to its effectiveness though as I have yet to reproduce. I think this may be win or qt/win specific. RJ, did your friend who saw this see it on Windows?

tl;dr is that if the sample rate the prefs load from the config is the same as the currently selected sample rate, the latencies aren't filled. This normally isn't a problem because the latency combo box population persists across openings and closings of the pref dialog but apparently this may only be true on Linux (or Linux+Max, or... but some combination not including Windows).

If no one can confirm this fix in the meantime I will try to repro and test the fix in win32 after exams.

**Comment by Holzhaus** [comment]
(by pegasus-renegadetech)
I saw this problem on Debian Linux AMD64 also, for the record. I haven't tested your possible fix yet. (To reproduce, just delete your config file.)

**Comment by Holzhaus** [comment]
(by bkgood)
This still doesn't break anything for me (pre-fix).

**Comment by Holzhaus** [comment]
(by pegasus-renegadetech)
Your fix works for me, though I hadn't deleted my config file before when it was happening. How does your code handle conflicts between old sound device settings in mixxx.cfg and ones in soundconfig.xml?

**Comment by Holzhaus** [comment]
(by bkgood)
It doesn't, it doesn't read anything from the global ConfigObject<ConfigValue>
passed around (it does set the sample rate because there are like 2 places in
mixxx that look for a sample rate there and I haven't gotten to stripping that
out yet).
